orderbook/src/api/get_trades.rs

```diff
@@ -0,0 +1,94 @@
+use super::H160Wrapper;
+use crate::api::convert_get_trades_error_to_reply;
+use crate::database::TradeFilter;
+use crate::orderbook::Orderbook;
+use anyhow::Result;
+use model::order::OrderUid;
+use model::trade::Trade;
+use serde::Deserialize;
+use std::{convert::Infallible, sync::Arc};
+use warp::{hyper::StatusCode, reply, Filter, Rejection, Reply};
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+struct Query {
+    pub order_uid: Option<OrderUid>,
+    pub owner: Option<H160Wrapper>,
+}
+
+impl Query {
+    fn trade_filter(&self) -> TradeFilter {
+        let to_h160 = |option: Option<&H160Wrapper>| option.map(|wrapper| wrapper.0);
+        TradeFilter {
+            order_uid: self.order_uid,
+            owner: to_h160(self.owner.as_ref()),
+        }
+    }
+}
+
+pub fn get_trades_request() -> impl Filter<Extract = (TradeFilter,), Error = Rejection> + Clone {
+    warp::path!("trades")
+        .and(warp::get())
+        .and(warp::query::<Query>())
+        .map(|query: Query| query.trade_filter())
+}
```

Comment:
Include request validation here that not both `owner` and `orderUid` are set. @fleupold I think this would satisfy the conversation we're having (here)[https://github.com/gnosis/gp-v2-services/pull/316#discussion_r580190577]